nrf_security/src/drivers/nrf_oberon/CMakeLists.txt

```diff
@@ -18,38 +18,38 @@ if(CONFIG_MBEDTLS_PSA_CRYPTO_C)
       ${NRFXLIB_ROOT}/crypto/nrf_oberon/include/mbedtls
       ${OBERON_PSA_CRYPTO_PATH}/oberon/drivers
   )
-
 endif()
 
-if (COMPILE_PSA_APIS)
-  append_with_prefix(src_crypto_oberon ${OBERON_PSA_CRYPTO_PATH}/oberon/drivers/
-    oberon_aead.c
-    oberon_cipher.c
-    oberon_ecdh.c
-    oberon_ecdsa.c
-    oberon_hash.c
-    oberon_helpers.c
-    oberon_kdf.c
-    oberon_key_pair.c
-    oberon_mac.c
-    oberon_rsa.c
-  )
-
-  append_with_prefix_ifdef(CONFIG_PSA_CRYPTO_DRIVER_ALG_CTR_DRBG_OBERON
-    src_crypto_oberon ${OBERON_PSA_CRYPTO_PATH}/oberon/drivers/
-      oberon_hmac_drbg.c
-  )
+macro(add_oberon_source kconfig filename)
+  if(CONFIG_PSA_CRYPTO_DRIVER_${kconfig}_OBERON)
+    list(APPEND
+      src_crypto_oberon
+      ${OBERON_PSA_CRYPTO_PATH}/oberon/drivers/${filename}
+	)
+  endif()
+endmacro()
 
-  append_with_prefix_ifdef(CONFIG_PSA_CRYPTO_DRIVER_ALG_HMAC_DRBG_OBERON
-    src_crypto_oberon ${OBERON_PSA_CRYPTO_PATH}/oberon/drivers/
-      oberon_ctr_drbg.c
-  )
+if (COMPILE_PSA_APIS)
+  # expands to
+  # CONFIG_PSA_CRYPTO_DRIVER_HAS_AEAD_SUPPORT_OBERON
+  add_oberon_source(HAS_AEAD_SUPPORT    oberon_aead.c)
+  add_oberon_source(HAS_KDF_SUPPORT     oberon_kdf.c)
+  add_oberon_source(HAS_ACCEL_KEY_TYPES oberon_key_pair.c)
+  add_oberon_source(HAS_MAC_SUPPORT     oberon_mac.c)
+  add_oberon_source(HAS_CIPHER_SUPPORT  oberon_cipher.c)
+  add_oberon_source(HAS_RSA_SUPPORT     oberon_rsa.c)
+  add_oberon_source(HAS_HASH_SUPPORT    oberon_hash.c)
+  add_oberon_source(ALG_ECDH            oberon_ecdh.c)
+  add_oberon_source(ALG_ECDSA           oberon_ecdsa.c)
+  add_oberon_source(ALG_CTR_DRBG        oberon_ctr_drbg.c)
+  add_oberon_source(ALG_HMAC_DRBG       oberon_hmac_drbg.c)
 
   # Add builtin functionality (if enabled)
   if(DEFINED src_crypto_oberon)
```

Comment:
oberon_helpers only added to the sources if atleast one more source is defined.
I find this, confusing, although it could be correct.

Maybe removed the check for list defined (`if (DEFINED src_crypto_oberon)`).
Or split this into two statements?

Reply:
I assumed oberon_helpers.c existed to help the other drivers here so the semantics would be desired.

"Maybe removed the check for list defined (if (DEFINED src_crypto_oberon))."

I assumed the if-statement was there for good reason, e.g. that it is possible to execute this code and end up with an empty list. Should I check this assumption and see if we can remove the if-statement?

"Or split this into two statements?"
This proposal I did not understand :(